MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1365,163 +1382,133 @@ std::string EnggDiffractionViewQtGUI::fittingPeaksData() const {
-    fittingRunNoChanged() {
-  // TODO: much of this should be moved to presenter
+void EnggDiffractionViewQtGUI::addBankItems(
+    std::vector<std::string> splittedBaseName, QString selectedFile) {
   try {
-    QString focusedFile = m_uiTabFitting.lineEdit_pushButton_run_num->text();
-    std::string strFocusedFile = focusedFile.toStdString();
-    // file name
-    Poco::Path selectedfPath(strFocusedFile);
-    Poco::Path bankDir;
-
-    // handling of vectors
-    m_fitting_runno_dir_vec.clear();
-    std::string strFPath = selectedfPath.toString();
-    std::vector<std::string> splitBaseName = splitFittingDirectory(strFPath);
-
-    if (selectedfPath.isFile() && !splitBaseName.empty()) {
-
-#ifdef __unix__
-      bankDir = selectedfPath.parent();
-#else
-      bankDir = (bankDir).expand(selectedfPath.parent().toString());
-#endif
-
-      if (!splitBaseName.empty() && splitBaseName.size() > 3) {
-        std::string foc_file = splitBaseName[0] + "_" + splitBaseName[1] + "_" +
-                               splitBaseName[2] + "_" + splitBaseName[3];
-        std::string strBankDir = bankDir.toString();
-        updateFittingDirVec(strBankDir, foc_file);
-      }
-      // if run number length greater
-    } else if (focusedFile.count() > 4) {
-      // if given a run number instead
-      updateFittingDirVec(m_focusDir, strFocusedFile);
-    } else {
-      userWarning("Invalid Input", "Invalid directory or run number given. "
-                                   "Please try again");
-    }
+    if (!m_fitting_runno_dir_vec.empty()) {
 
-    try {
-      // add bank to the combo-box and list view
-      addBankItems(splitBaseName, focusedFile);
-    } catch (std::runtime_error &re) {
-      userWarning("Unable to insert items: ",
-                  "Could not add banks to "
-                  "combo-box or list widget; " +
-                      static_cast<std::string>(re.what()) +
-                      ". Please try again");
-    }
-  } catch (std::runtime_error &re) {
-    userWarning("Invalid file", "Unable to select the file; " +
-                                    static_cast<std::string>(re.what()));
-    return;
-  }
-}
+      // delete previous bank added to the list
+      m_uiTabFitting.comboBox_bank->clear();
 
-void EnggDiffractionViewQtGUI::updateFittingDirVec(std::string &bankDir,
-                                                   std::string &focusedFile) {
+      for (size_t i = 0; i < m_fitting_runno_dir_vec.size(); i++) {
+        Poco::Path vecFile(m_fitting_runno_dir_vec[i]);
+        std::string strVecFile = vecFile.toString();
+        // split the directory from m_fitting_runno_dir_vec
+        std::vector<std::string> vecFileSplit =
+            splitFittingDirectory(strVecFile);
 
-  try {
+        // get the last split in vector which will be bank
+        std::string bankID = (vecFileSplit[vecFileSplit.size() - 1]);
+
+        bool digit = isDigit(bankID);
```

Comment:
The presenter does the checks and passes the values ready to the view, whatever it needs here it should be calculated outside of the view.